appeals/web/src/server/lib/page-components/boolean.js

```diff
@@ -0,0 +1,49 @@
+import { convertFromBooleanToYesNo } from '#lib/boolean-formatter.js';
+import { mapActionComponent } from '#lib/mappers/component-permissions.mapper.js';
+import { permissionNames } from '#environment/permissions.js';
+
+/**
+ *
+ * @param {Object} options
+ * @param {string} options.id
+ * @param {string} options.text
+ * @param {boolean|null} [options.value]
+ * @param {string} [options.defaultText]
+ * @param {string} options.link
+ * @param {boolean} [options.addCyAttribute]
+ * @param {import("express-session").Session & Partial<import("express-session").SessionData>} options.session
+ * @returns {Instructions}
+ */
+export function booleanDisplayField({
+	id,
+	text,
+	value,
+	defaultText,
+	link,
+	addCyAttribute,
+	session
+}) {
+	return {
```

Comment:
Would it be more flexible to create a helper function to generate summaryListItems, and have data-type-specific functions which call that, eg. for boolean values in this case?

Reply:
Possibly, I think the duplicate code between the summary list items is very minimal. Might be worth exploring though.

Reply:
Have you seen Lucy's PR https://github.com/Planning-Inspectorate/appeals-back-office/pull/432? there are plenty of examples in there of instances where a summaryListItem helper would save quite a bit of code (this would also reduce errors in the same way as the `yesNoInput` helper you added recently). That would be one for a separate PR regardless.